### PR TITLE
#269 : Missing Management API Functionality for Linking User Accounts…

### PR DESCRIPTION
**Pull request by cschwalm**
### Changes

The Java Auth0 API is missing the ability to link accounts using the "link_with" body parameter option. This is useful for when the client sends the secondary ID token to link with.

Override the linkIdentity method in UsersEntity.java to support the ability to supply a ID token instead of a connection and provider string. 

### References

https://github.com/auth0/auth0-java/issues/269
See: https://auth0.com/docs/api/management/v2#!/Users/post_identities

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors. 

- [ X] This change adds test coverage
- [ X] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [X ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [ X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [ X] All existing and new tests complete without errors

### CODE REVIEW

File: src/main/java/com/auth0/client/mgmt/UsersEntity.java [hunk 1]
```diff
@@ -337,6 +337,35 @@ public Request deleteMultifactorProvider(String userId, String provider) {
         return request;
     }
 
+    /**
+     * Links two User's Identities via token using the link_with parameter.
+     * A token with scope update:users is needed or update:current_user_identities if access token is for same user.
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/UsersEntity.java:342]: According to the API docs, the `link_with` approach will only work for `update:current_user_identities` scoped tokens. That also means that the API client would need to be set up with that token. I think the use case of these scopes and particularly this endpoint is more related to native clients. But I don't have any issues with adding it. 

Let's fix that doc

```suggestion
     * A token with scope update:current_user_identities is needed. It only works for the user the access token represents.
```

File: src/test/java/com/auth0/client/mgmt/UsersEntityTest.java [hunk 2]
```diff
@@ -574,6 +574,13 @@ public void shouldThrowOnLinkUserIdentityWithNullSecondaryId() throws Exception
         api.users().linkIdentity("1", null, "auth0", null);
     }
 
+    @Test
+    public void shouldThrowOnLinkUserIdentityWithNullSecondaryToken() throws Exception {
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/client/mgmt/UsersEntityTest.java:578]: Let's repeat this one for the user id, which also needs to be set.

File: src/main/java/com/auth0/client/mgmt/UsersEntity.java [hunk 3]
```diff
@@ -337,6 +337,35 @@ public Request deleteMultifactorProvider(String userId, String provider) {
...
+     * See https://auth0.com/docs/api/management/v2#!/Users/post_identities
+     *
+     * @param primaryUserId   the primary identity's user id
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/UsersEntity.java:345]:
```suggestion
     * @param primaryUserId   the primary identity's user id associated with the access token this client was configured with.
```

File: src/main/java/com/auth0/client/mgmt/UsersEntity.java [hunk 4]
```diff
@@ -337,6 +337,35 @@ public Request deleteMultifactorProvider(String userId, String provider) {
...
+     * @param secondaryIdToken the user identity token (JWT)
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/UsersEntity.java:346]:
```suggestion
     * @param secondaryIdToken the user ID token representing the identity to link with the current user
```

File: src/main/java/com/auth0/client/mgmt/UsersEntity.java (hunk 1 above)
Comment (cschwalm) [reply src/main/java/com/auth0/client/mgmt/UsersEntity.java]: Added. No problem!

File: src/main/java/com/auth0/client/mgmt/UsersEntity.java (hunk 4 above)
Comment (cschwalm) [reply src/main/java/com/auth0/client/mgmt/UsersEntity.java]: Added.

File: src/test/java/com/auth0/client/mgmt/UsersEntityTest.java (hunk 2 above)
Comment (cschwalm) [reply src/test/java/com/auth0/client/mgmt/UsersEntityTest.java:578]: Added. Sorry for the miss. Good catch.

File: src/test/java/com/auth0/client/mgmt/UsersEntityTest.java [hunk 5]
```diff
@@ -574,6 +574,20 @@ public void shouldThrowOnLinkUserIdentityWithNullSecondaryId() throws Exception
...
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'secondary id token' cannot be null!");
+        api.users().linkIdentity("1", null);
+    }
+
+    @Test
+    public void shouldThrowOnLinkUserIdentityWithNullUserId() throws Exception {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'primary user id' cannot be null!");
+        api.users().linkIdentity("1", null);
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/client/mgmt/UsersEntityTest.java:588]: this should fix the tests
```suggestion
        api.users().linkIdentity(null, "2");
```

**Pull request merged**